src/index.js

```diff
@@ -95,6 +99,7 @@ export default function supernova(env) {
           });
         }
       }, [
+        rootElement,
```

Comment:
If anything, this should have the reactRoot as dependency, otherwise we could still be passing the old reactRoot

Reply:
good find. I mistakenly remove the line.

Reply:
you wont need rootElement though since if `rootElement` changes `reactRoot` will change too

Reply:
that's true. Updated.